src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -52,15 +57,27 @@ private SwitchBootstraps() {}
 
     private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
 
-    private static final MethodHandle DO_TYPE_SWITCH;
-    private static final MethodHandle DO_ENUM_SWITCH;
+    private static final MethodHandle INSTANCEOF_CHECK;
+    private static final MethodHandle INTEGER_EQ_CHECK;
+    private static final MethodHandle OBJECT_EQ_CHECK;
+    private static final MethodHandle NULL_CHECK;
+    private static final MethodHandle IS_ZERO;
+    private static final MethodHandle ENUM_LOOKUP;
 
     static {
         try {
-            DO_TYPE_SWITCH = LOOKUP.findStatic(SwitchBootstraps.class, "doTypeSwitch",
-                                           MethodType.methodType(int.class, Object.class, int.class, Object[].class));
-            DO_ENUM_SWITCH = LOOKUP.findStatic(SwitchBootstraps.class, "doEnumSwitch",
-                                           MethodType.methodType(int.class, Enum.class, int.class, Object[].class));
+            INSTANCEOF_CHECK = LOOKUP.findStatic(SwitchBootstraps.class, "instanceofCheck",
+                                           MethodType.methodType(boolean.class, Object.class, Class.class));
```

Comment:
I am tempted to have
```suggestion
            INSTANCEOF_CHECK = MethodHandles.permuteArguments(LOOKUP.findVirtual(Class.class, "isInstance",
                                           MethodType.methodType(boolean.class, Object.class)), MethodType.methodType(boolean.class, Object.class, Class.class), 1, 0);
```
to reduce the implementation methods in SwitchBootstraps